samtranslator/model/eventsources/pull.py

```diff
@@ -188,6 +188,13 @@ def _validate_filter_criteria(self):
         if list(self.FilterCriteria.keys()) not in [[], ["Filters"]]:
             raise InvalidEventException(self.relative_id, "FilterCriteria field has a wrong format")
 
+    def validate_secrets_manager_kms_key_id(self):
```

Comment:
can you make it private method, and do the validation only if the SecretsManagerKmsKeyId is not None.

Reply:
the validation is called only when `SecretsManagerKmsKeyId` is provided. It is not a required property for an eventsource type